internal/wasm/sdk/opa/opa_bench_test.go

```diff
@@ -0,0 +1,194 @@
+	input := make(map[string]interface{})
+
+	for i := 0; i < b.N; i++ {
+		if _, err := pq.Eval(ctx, rego.EvalInput(input)); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func BenchmarkWASMArrayIteration(b *testing.B) {
+	sizes := []int{10, 100, 1000, 10000}
+	for _, n := range sizes {
+		b.Run(fmt.Sprint(n), func(b *testing.B) {
+			benchmarkIteration(b, test.ArrayIterationBenchmarkModule(n))
+		})
+	}
+}
+
+func BenchmarkWASMSetIteration(b *testing.B) {
+	sizes := []int{10, 100, 1000, 10000}
+	for _, n := range sizes {
+		b.Run(fmt.Sprint(n), func(b *testing.B) {
+			benchmarkIteration(b, test.SetIterationBenchmarkModule(n))
+		})
+	}
+}
+
+func BenchmarkWASMObjectIteration(b *testing.B) {
+	sizes := []int{10, 100, 1000, 10000}
+	for _, n := range sizes {
+		b.Run(fmt.Sprint(n), func(b *testing.B) {
+			benchmarkIteration(b, test.ObjectIterationBenchmarkModule(n))
+		})
+	}
+}
+
+var r *opa.Result
+
+func benchmarkIteration(b *testing.B, module string) {
+	query := "data.test.main = x"
+	policy := compileRegoToWasm(module, query, false)
+
+	instance, err := opa.New().
+		WithPolicyBytes(policy).
+		WithMemoryLimits(2*wasm.PageSize, 47*wasm.PageSize).
+		WithPoolSize(1).
+		Init()
+	if err != nil {
+		b.Fatalf("init sdk: %v", err)
+	}
+
+	b.ResetTimer()
+	ctx := context.Background()
+	var input interface{} = make(map[string]interface{})
+
+	for i := 0; i < b.N; i++ {
+		r, err = instance.Eval(ctx, opa.EvalOpts{Input: &input})
+		if err != nil {
+			b.Fatalf("Unexpected query error: %v", err)
+		}
+		if string(r.Result) != `{{"x":true}}` {
+			b.Errorf("unexpected result: %s", string(r.Result))
+		}
+	}
+}
+
+func BenchmarkWASMLargeJSON(b *testing.B) {
+	for _, kv := range []struct{ key, val int }{
+		{10, 10},
+		{10, 100},
+		{10, 1000},
+		{10, 10000},
+		{100, 100},
+		{100, 1000},
+	} {
+		b.Run(fmt.Sprintf("%dx%d", kv.key, kv.val), func(b *testing.B) {
+			ctx := context.Background()
+			data := test.GenerateJSONBenchmarkData(kv.key, kv.val)
+
+			// Read data.values N times inside query.
+			query := "data.keys[_] = x; data.values = y"
+			policy := compileRegoToWasm("", query, false)
+
+			instance, err := opa.New().
+				WithPolicyBytes(policy).
+				WithDataJSON(data).
+				WithMemoryLimits(200*wasm.PageSize, 600*wasm.PageSize). // This is rather much
+				WithPoolSize(1).
+				Init()
+			if err != nil {
+				b.Fatalf("init sdk: %v", err)
+			}
+
+			b.ResetTimer()
+			var input interface{} = make(map[string]interface{})
+
+			for i := 0; i < b.N; i++ {
+				r, err = instance.Eval(ctx, opa.EvalOpts{Input: &input})
+				if err != nil {
+					b.Fatalf("Unexpected query error: %v", err)
+				}
+			}
+		})
+	}
+}
+
+func BenchmarkWASMVirtualDocs(b *testing.B) {
+	for _, kv := range []struct{ total, hit int }{
+		{1, 1},
+		{10, 1},
+		{100, 1},
+		{1000, 1},
+		{10, 10},
+		{100, 10},
+		{1000, 10},
+		{100, 100},
+		{1000, 100},
+		{1000, 1000},
+	} {
+		b.Run(fmt.Sprintf("total=%d/hit=%d", kv.total, kv.hit), func(b *testing.B) {
```

Comment:
I _much_ prefer this notation to the `NxM` style used in the topdown tests...I can never remember what the topdown test names refer to without looking at the implementation 😅

Reply:
We could fix the other ones to use this style, we'd only loose one comparison in the bench diff stuff we have (due to the renamed tests).